src/html_tag.cpp

```diff
@@ -2364,6 +2364,54 @@ bool litehtml::html_tag::set_pseudo_class( const tchar_t* pclass, bool add )
 	return ret;
 }
 
+bool litehtml::html_tag::set_class( const tchar_t* pclass, bool add )
+{
+	bool ret = false;
+	string_vector tokens;
+	split_string(m_class, tokens, _t(" "));
+
+	if(add)
+	{
+		for(string_vector::iterator tok = tokens.begin(); tok != tokens.end(); tok++)
+		{
+			if(*tok == pclass)
+			{
+				return false;
+			}
+		}
+
+		m_class += _t(" ");

```

Comment:
this will potentially prepend a space to the m_class variable ( if it was empty before )